### PR TITLE
gh-103088: fix activate script not working in Cygwin

### PR DESCRIPTION
**Pull request by Zabolekar**
* Issue: gh-103088

### CODE REVIEW

**Comment by zooba** [comment]
Can we add a test to the `test_venv` module for this?

Basically confirm that `\r\n` does not exist in the `activate` Bash script on any platform after a venv is created.

**Comment by Zabolekar** [comment]
> Can we add a test to the `test_venv` module for this?

Test added.

File: Lib/test/test_venv.py [hunk 1]
```diff
@@ -611,6 +611,20 @@ def test_zippath_from_non_installed_posix(self):
         out, err = check_output(cmd)
         self.assertTrue(zip_landmark.encode() in out)
 
+    def test_activate_shell_script_has_no_dos_newlines(self):
+        """
+        Test that the `activate` shell script contains no CR LF.
+        This is relevant for Cygwin.
```
Comment (zooba) [inline Lib/test/test_venv.py]:
```suggestion
        This is relevant for Cygwin, as the Windows build might have
        converted line endings accidentally.
```

File: Misc/NEWS.d/next/Windows/2023-04-12-10-49-21.gh-issue-103088.Yjj-qJ.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+fix activate script not working in Cygwin
```
Comment (zooba) [inline Misc/NEWS.d/next/Windows/2023-04-12-10-49-21.gh-issue-103088.Yjj-qJ.rst]:
```suggestion
Fix virtual environment :file:`activate` script having incorrect line endings for Cygwin.
```

**Review by zooba — COMMENTED** [review]
Minor doc tweaks, I'll commit them myself and merge.

**Comment by zooba** [comment]
Buildbot failures are being discussed in the bug #103088

**Pull request merged**